tests/python/test_schedule_ops.py

```diff
@@ -22,6 +35,29 @@ def is_pre_volta():
     return prop.major < 7
 
 
+# A helper function to test heuristic schedulers with user schedules
+def _apply_scheduler_helper(schedule, selected_heuristic):
+    # Check that only selected heuristic is available as a scheduler
+    available_heuristics = schedule.find_compatible_schedulers()
+    assert available_heuristics == [selected_heuristic]
+
+    # Double-check with can_schedule
+    status, _ = schedule.can_schedule(selected_heuristic)
+    assert status
+
+    # Check that the other schedulers are not compatible with this fusion
+    assert all(
+        [
+            not schedule.can_schedule(h)[0]
```

Comment:
Is this expected?

On the cpp side, I think we can have multiple executor being able to compile the same fusion. i.e. no-op and pointwise have overlap when alias analysis is run successfully.
We cannot run preseg passes yet, so I don't have a good counter example right now. But I'm curious what's your motivation to assert here.

Reply:
The implicit assumption is that only one scheduler will work in the `auto_scheduler` tests. `_apply_scheduler_helper` checks for this specifically.